tests/pos/specialized-traits-strawman.scala

```diff
@@ -0,0 +1,142 @@
+import language.experimental.erasedDefinitions
+
+// Source code
+
+/*inline*/ trait Iterator[T/*: Specialized*/]:
+  def hasNext: Boolean
+  def next(): T
+
+/*inline*/ trait ArrayIterator[T/*: Specialized*/](elems: Array[T]) extends Iterator[T]:
+  private var current = 0
+  def hasNext: Boolean = current < elems.length
+  def next(): T = try elems(current) finally current += 1
+
+/*inline*/ trait Iterable[T/*: Specialized*/]:
+  def iterator: Iterator[T]
+  def forall(f: T => Unit): Unit =
+    val it = iterator
+    while it.hasNext do f(it.next())
+
+/*inline*/ trait Seq[T/*: Specialized*/](elems: Array[T])
+extends Iterable[T]:
+  def length: Int           = elems.length
+  def apply(i: Int): T      = elems(i)
+  def iterator: Iterator[T] = new ArrayIterator[T](elems) {}
+
+// Specialized trait APIs generated from specialization:
+
+trait Iterator_Int extends Iterator[Int]:
+  def hasNext: Boolean
+  def next(): Int
+
+trait ArrayIterator_Int extends ArrayIterator[Int], Iterator[Int]
```

Comment:
@odersky I've just tried it with 3.5.0 and it compiles fine with the following parents. But we agree that after erasure, we should have `Iterator_Int` anyways.
```suggestion
trait ArrayIterator_Int extends ArrayIterator[Int], Iterator_Int
```

Reply:
I believe it does not work if you replace _both_ uses, the one here and the one that would map `Seq[Int]` to `Seq_Int`.